src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/LeftAlignAndTrimVariants.java

```diff
@@ -352,127 +273,56 @@ protected int trimAlign(final VariantContext vc, final ReferenceContext ref) {
      * @param ref Reference context
      * @return new VC.
      */
-    protected VariantContext leftAlign(final VariantContext vc, final ReferenceContext ref) {
-        if (!vc.isSimpleIndel()) {
+    @VisibleForTesting
+    static VariantContext leftAlignAndTrim(final VariantContext vc, final ReferenceContext ref, final int maxLeadingBases, final boolean trim) {
+        if (!vc.isIndel() || maxLeadingBases <= 0) {
             return vc;
         }
 
-        // get the indel length
-        final int indelLength = Math.abs(vc.getIndelLengths().get(0));
+        for(int leadingBases = Math.min(maxLeadingBases, 10); leadingBases <= maxLeadingBases; leadingBases *= 2) {
```

Comment:
I believe there is a bug here (quite possibly originating prior to this PR).  If `maxLeadingBases` is greater than 10, but not a multiple of 10, there can be problems.  Suppose `maxLeadingBases=15`, and the indel should be left aligned that full amount.  `leadingBases` will initialize to 10, and the indel will left align the full 10 bases.  `leadingBases` will then double to 20, the for loop condition will be false, and the variant won't be shifted left at all.  I think this is easy to fix, just needs a slight adjustment to the control flow logic.

Also would be nice to have a test of this type of case, to ensure it is handled correctly.

Reply:
done